src/main/java/org/edumips64/MainCLI.java

```diff
@@ -62,146 +67,46 @@ public static void main(String args[]) {
-              } else {
-                System.out.println(c.getRegister(Integer.parseInt(tokens[2])));
-              }
-            } else if (tokens[1].compareToIgnoreCase("memory") == 0) {
-              System.out.println(memory);
-            } else if (tokens[1].compareToIgnoreCase("symbols") == 0) {
-              System.out.println(symTab);
-            } else if (tokens[1].compareToIgnoreCase("pipeline") == 0) {
-              System.out.println(c.pipeLineString());
-            } else {
-              System.out.println("The show command requires at least one parameter.");
-            }
-          }
-        } else if (tokens[0].compareTo("run") == 0) {
-            int steps = 0;
-            long startTimeMs = System.currentTimeMillis();
-            try {
-              while(true) {
-                steps++;
-                c.step();
-
-              }
-            } catch (HaltException e) {
-              long endTimeMs = System.currentTimeMillis();
-              long totalTimeMs = endTimeMs - startTimeMs;
-              System.out.println("Execution ended. " + steps + " steps were executed in " + totalTimeMs + "ms");
-            }
-        } else if (tokens[0].compareTo("step") == 0) {
-          try {
-            int num = 1;
+      runReplLoop(commandLine);
 
-            if (tokens.length > 1) {
-              num = Integer.parseInt(tokens[1]);
-            }
-
-            try {
-              if (num > 0) {
-                System.out.println("I execute " + num + " steps of simulation.");
-              }
+    } catch (Exception e) {
+      Cli.printErrorMessage(e);
+      System.exit(1);
+    }
+  }
 
-              for (int i = 0; i < num; ++i) {
-                c.step();
-                System.out.println(c.pipeLineString());
-              }
-            } catch (Exception e) {
-              System.err.println(getErrorReportingMessage());
-              e.printStackTrace();
-            }
-          } catch (NumberFormatException e) {
-            System.out.println("The second parameter of the step command must be an integer.");
-          }
-        } else {
-          System.out.println("Unknown command.\nType 'help' to get the list of available commands.");
+  private static void runReplLoop(CommandLine commandLine) throws IOException {
+      BufferedReader keyboard = new BufferedReader(new InputStreamReader(System.in));
+      printArrow();
+      while (true) {
+          String read = keyboard.readLine();
+          handleTokens(commandLine, read.split(" "));
+          printArrow();
         }
+    }
 
-        System.out.print("> ");
-      }
+  private static void printArrow() {
+    System.out.print("> ");
+  }
 
-      System.out.println("Bye!");
-    } catch (Exception e) {
-      System.err.println(getErrorReportingMessage());
-      e.printStackTrace();
-      System.exit(1);
+  private static void handleTokens(CommandLine commandLine, String[] tokens) {
+    if (tokens.length > 0) {
```

Comment:
I tried the last JAR and it still seems to show the error before showing the usage:
![image](https://user-images.githubusercontent.com/200139/92300386-5162af00-ef52-11ea-9ecd-8ea5ac867e06.png)

Maybe `tokens` here is not empty but has empty/null entries?

This is definitely a minor thing though. I wouldn't block merging on this one.